packages/gatsby/cache-dir/gatsby-browser-entry.js

```diff
@@ -1,4 +1,4 @@
-import React from "react"
+import React, { useContext } from "react"
```

Comment:
This isn't a functional change necessarily, but maybe just _use_ `React.useContext` instead? We can check for the existence of that rather easily, e.g.

```jsx
if (typeof React.useContext !== `function` && process.env.NODE_ENV === `development`) {
  // warn
}
```

Reply:
Also one note here... I think it's worth validating that these checks are stripped from our code in production (e.g. with our uglifier).

Reply:
This is tricky, because I would want this error to print out during `built-html` stage (when we write to static html files), but not actually end up in client js bundle (just micro optim to save few bytes)

Reply:
@pieh This will never end up in the bundle because I'll be stripping away the import and replacing it with static data

Reply:
All imports to useStaticQuery are now removed in build so this should be good

Reply:
Ah, you are completely right - so this check would need to be done in babel plugin? (yuck - mixed concerns)

Reply:
Actually, I guess it would need to be duplicated - in case there are 3rd party tools (Storybook etc) that want to use `gatsby-browser-entry` outside of `gatsby` "realm"?